src/sentry/static/sentry/app/utils/tokenizeSearch.tsx

```diff
@@ -203,8 +211,17 @@ export class QueryResults {
         const token = this.tokens[i];
         const prev = this.tokens[i - 1];
         const next = this.tokens[i + 1];
-        if (isOp(token) && ['OR', 'AND'].includes(token.value)) {
+        if (isOp(token) && isBooleanOp(token.value)) {
           if (prev === undefined || isOp(prev) || next === undefined || isOp(next)) {
+            // Want to avoid removing `(term) OR (term)`
+            if (
```

Comment:
wondering if this can be simplified to be `isOpenParen(next) && isClosingParen(prev)`?

Reply:
is that more simple if this is the only place we use those functions? If we need more special casing I think we could refactor this then

Reply:
I think the if guard seems redundant in checking for the token values.

Maybe this could be better?

```
function isParen(token: Token, character: '(' | ')') {
  return token !== undefined && isOp(token) && ['(', ')'].includes(token.value) && token.value === character;
}

//...

if(isParen(prev, ')') && isParen(next, '(')) {  
//...
```

Reply:
Ahhh sorry I thought you were suggesting two new functions, that's a much better approach :+1: